metricbeat/beater/module.go

```diff
@@ -0,0 +1,204 @@
+package beater
+
+import (
+	"expvar"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/filter"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/publisher"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/joeshaw/multierror"
+	"github.com/pkg/errors"
+)
+
+// Expvar metric names.
+const (
+	successesKey = "success"
+	failuresKey  = "failures"
+	eventsKey    = "events"
+)
+
+var (
+	debugf  = logp.MakeDebug("metricbeat")
+	fetches = expvar.NewMap("fetches")
+)
+
+// moduleWrapper contains the Module and the private data associated with
+// running the Module and its MetricSets. The moduleWrapper contains a list
+// of pointer to its associated MetricSets.
+type moduleWrapper struct {
+	mb.Module
+	filters   *filter.FilterList
+	pubClient publisher.Client
+	children  []*metricSetWrapper
```

Comment:
I was first confused by this name as it sounds like "children modules", but it is actually metricsets.